python/arcticdb/version_store/_normalization.py

```diff
@@ -1040,7 +1054,12 @@ def _custom_pack(self, obj):
     def _ext_hook(self, code, data):
         if code == MsgPackSerialization.PD_TIMESTAMP:
             data = MsgPackNormalizer._nested_msgpack_unpackb(data)
-            return pd.Timestamp(data[0], tz=data[1]) if data[1] is not None else pd.Timestamp(data[0])
+            val, tz, offset = data
```

Comment:
It may be wise to keep using the numeric indexing so that we can continue to extend if needed.

Also this needs to be able to read old data which only had two fields.   So we should add a test that covers the legacy case as well.
```
if len(data) == 2:
  val, tz = data
  # old method...
else:
  val, tz, offset = data[:3]
  # new methods...
```

Reply:
Indeed I had forgotten to push these changes.
As for testing the legacy case, we can do it only with the compatibility tests and I have added the needed changes [here](https://github.com/man-group/ArcticDB/pull/1407/files#diff-4df7402f5282307e8aa7d22634403c683d6ce0c08840a40eba4c47f12665b008)

Reply:
How does the compatibility test change result in a test that exercises the `if len(data)==2:` block?